proxy.h

```diff
@@ -558,6 +562,44 @@ class proxy : public details::facade_traits<F>::base {
   static constexpr bool HasMoveAssignment = HasMoveConstructor && HasDestructor;
 
  public:
+  template <class O>
+  class dispatch_ptr {
+    using OverloadTraits = details::overload_traits<O>;
+    using DispatcherType = typename OverloadTraits::dispatcher_type;
+
+   public:
+    constexpr dispatch_ptr() noexcept = default;
+    constexpr dispatch_ptr(const dispatch_ptr&) noexcept = default;
+    dispatch_ptr& operator=(const dispatch_ptr&) noexcept = default;
+
+    template <class D>
+    constexpr explicit dispatch_ptr(std::in_place_type_t<D>) noexcept
+#if defined(_MSC_VER) && !defined(__clang__)
```

Comment:
put the full name of ctor into this #if block?

Reply:
Done